### ISSUE TITLE
Crash in my app

### ISSUE THREAD

**Issue by UrielFrankel** [centrifugal/centrifuge-java#67]
Here is the stack trace:
```
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.FutureTask@651c9a5[Not completed] rejected from java.util.concurrent.ThreadPoolExecutor@6e1517a[Shutting down, pool size = 1, active threads = 1, queued tasks = 2, completed tasks = 1]
	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2072)
	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:834)
	at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1364)
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:118)
	at java.util.concurrent.Executors$DelegatedExecutorService.submit(Executors.java:671)
	at io.github.centrifugal.centrifuge.Client$1.onClosed(SourceFile:15)
	at okhttp3.internal.ws.RealWebSocket.writeOneFrame$okhttp(SourceFile:188)
	at okhttp3.internal.ws.RealWebSocket$WriterTask.runOnce(SourceFile:3)
	at okhttp3.internal.concurrent.TaskRunner.runTask(SourceFile:63)
	at okhttp3.internal.concurrent.TaskRunner.access$runTask(Unknown Source:0)
	at okhttp3.internal.concurrent.TaskRunner$runnable$1.run(SourceFile:55)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
	at java.lang.Thread.run(Thread.java:1012)
```

I have zero clue what happen. As you can see I can't even guard it with try/catch. I called unsubscribe() and then close(). Is that the problem? Should I put the close() in the onDestroy of the App? thanks.

**Comment by FZambia** [comment]
Hello @UrielFrankel - seems that you can easily reproduce this? Can you provide example code?

**Comment by UrielFrankel** [comment]
I can't reproduce it but I can send you what I do:

    fun disconnect() {
        unsubscribe()
        Logger.d(TAG, "disconnect() called")
        try {
            client?.close(5000) // from demo app
            client = null
        } catch (e: InterruptedException) {
            Logger.e(TAG, "error in centrifugo disconnection ", e)
        }
        this.connectionId = null
    }

    fun unsubscribe() {
        Logger.d(TAG, "unsubscribe() called")
        currentSubscription?.let {
            client?.removeSubscription(it)
            currentSubscription = null
        }
    }

I suspect that calling removeSubscription and immediately close is the problem

**Comment by FZambia** [comment]
Thanks, I'll experiment - in general you need to call Close when you don't need client anymore. After calling Close client is unusable.

But in this case I suppose `onClosed` callback of WebSocket is called after ThreadPoolExecutor already closed. We already handle RejectedExecutionException in OnFailure - https://github.com/centrifugal/centrifuge-java/blob/f11c20c54e9a0a91af44c4597a005ce2a2cd40c7/centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java#L378, but for some reason not handling it in other WebSocket callbacks - like the one in your stacktrace (`onClosed`).

**Comment by UrielFrankel** [comment]
Good news, I was able to reproduce it in my app. I called for close before i was fully connected and close calls disconnect.

**Comment by FZambia** [comment]
Reproduced locally and opened #71

**Comment by FZambia** [comment]
The fix is shipped in [0.3.1](https://github.com/centrifugal/centrifuge-java/releases/tag/0.3.1)
